flink-core/src/main/java/org/apache/flink/api/connector/sink2/StatefulSink.java

```diff
@@ -36,49 +34,43 @@
  * @param <WriterStateT> The type of the sink writer's state
  */
 @PublicEvolving
-public interface StatefulSink<InputT, WriterStateT> extends Sink<InputT> {
+@Deprecated
+public interface StatefulSink<InputT, WriterStateT>
+        extends Sink<InputT>, SupportsWriterState<InputT, WriterStateT> {
 
     /**
-     * Create a {@link StatefulSinkWriter}.
+     * Create a {@link org.apache.flink.api.connector.sink2.StatefulSinkWriter} from a recovered
+     * state.
      *
      * @param context the runtime context.
      * @return A sink writer.
      * @throws IOException for any failure during creation.
      */
-    StatefulSinkWriter<InputT, WriterStateT> createWriter(InitContext context) throws IOException;
```

Comment:
is it okay to remove this method? Technically there could be scenarios where this would be a breaking change.

Reply:
`StatefulSink` still inherits from `Sink` which has this deprecated method which will take its place:
```
    @Deprecated
    SinkWriter<InputT> createWriter(InitContext context) throws IOException;
```

So I think removing this is not a breaking change. @gyfora: Do I miss something?

Thanks,
Peter

Reply:
except if someone directly uses the `createWriter` method of the `StatefulSink` class and expects it to return the `StatefulSinkWriter` subclass instead of `SinkWriter` right?

Reply:
not sure if this is a big issue but we should at least make a note of it

Reply:
As this is a required change to move forward and was accepted as part of the Flip, it is good to go from my end.
I don't expect this to cause any problems in the Flink / connector ecosystem